python/cudf/cudf/core/udf/masked_lowering.py

```diff
@@ -333,6 +333,27 @@ def masked_scalar_cast_impl(context, builder, sig, args):
     return result._getvalue()
 
 
+@cuda_lower(abs, MaskedType)
+def masked_scalar_abs_impl(context, builder, sig, args):
+    input = cgutils.create_struct_proxy(sig.args[0])(
+        context, builder, value=args[0]
+    )
+    result = cgutils.create_struct_proxy(sig.return_type)(context, builder)
+
+    result.valid = input.valid
+    with builder.if_then(input.valid):
+        result.value = context.compile_internal(
+            builder,
+            lambda x: abs(x),
```

Comment:
Is it possible to just write `abs` here ($\eta$-reduction)

Reply:
It might not be. I would defer to @brandon-b-miller here but I remember raising a similar question someplace else with Numba.

Reply:
what @bdice mentions is correct. We need to wrap the actual `abs` call in an expression that as a whole exposes its bytecode as a `__code__` attribute. Switching to `abs` results in the following:
```python
'builtin_function_or_method' object has no attribute '__code__'
```